curriculum/service/app.py

```
@@ -0,0 +1,119 @@
+@app.get("/api/ai/curriculum/catalog")
+def catalog():
+    """로드맵이 고를 수 있는 강의 목록. 화면에서 트랙·난이도를 보여줄 때 쓴다."""
+    return {"count": len(COURSES), "courses": COURSES}
+
+
+@app.post("/api/ai/curriculum/roadmap", response_model=RoadmapResponse)
+def create_roadmap(req: RoadmapRequest):
+    try:
+        llm = roadmap.Gemini()
+    except roadmap.LLMError as e:
+        raise HTTPException(status_code=503, detail=f"모델을 부를 수 없다: {e.detail}")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)